charts/src/main/java/hu/ma/charts/line/LineChart.kt

```diff
@@ -0,0 +1,363 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.nativeCanvas
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Dp
+import hu.ma.charts.legend.DrawHorizontalLegend
+import hu.ma.charts.legend.DrawVerticalLegend
+import hu.ma.charts.line.data.DrawAxis
+import hu.ma.charts.line.data.LineChartData
+import hu.ma.charts.legend.LegendPosition
+import hu.ma.charts.pie.createLegendEntries
+import kotlin.math.abs
+import kotlin.math.max
+
+@Composable
+fun LineChart(
+  chartHeight: Dp? = null,
+  data: LineChartData,
+  onDrillDown: ((x: Int, series: List<LineChartData.SeriesData>) -> Unit)? = null,
+) {
+
+  val legendEntries = remember(data) { data.createLegendEntries(data.legendShapeSize) }
+
+  @Composable
+  fun legend() {
+    when (data.legendPosition) {
+      LegendPosition.End, LegendPosition.Start -> {
+        DrawVerticalLegend(legendEntries)
+      }
+      LegendPosition.Top, LegendPosition.Bottom ->
+        DrawHorizontalLegend(legendEntries)
+    }
+  }
+
+  val maxXValues = data.series.maxOf { it.points.maxOf { point -> point.x } }
+
+  val maxYValue = max(
+    data.series.maxOf { it.points.maxOf { point -> point.value } },
+    if (data.yLabels.isNotEmpty()) data.yLabels.maxOf { it.atValue } else 0f
+  )
+
+  var drillDownPoint by remember { mutableStateOf<Float?>(null) }
+
+  BoxWithConstraints(modifier = Modifier.fillMaxSize()) {
+    val maxWidth = this.maxWidth
+
+    Column {
+      if (data.legendPosition == LegendPosition.Top) {
+        Row(
+          modifier = Modifier.fillMaxWidth(),
+          horizontalArrangement = data.legendAlignment.toHorizontalArrangement()
+        ) {
+          legend()
+        }
+        Spacer(modifier = Modifier.requiredSize(data.legendOffset))
+      }
+
+      val animatedDrillDownX by animateFloatAsState(
+        targetValue = drillDownPoint ?: with(LocalDensity.current) { maxWidth.toPx() }
+      )
+
+      Row(
+        modifier = Modifier.wrapContentHeight(),
+        verticalAlignment = data.legendAlignment.toVerticalAlignment()
+      ) {
+
+        if (data.legendPosition == LegendPosition.Start) {
+          Column(
+            modifier = Modifier
+              .wrapContentWidth()
+              .padding(end = data.legendOffset)
+          ) {
+            legend()
+          }
+        }
+
+        var modifier = Modifier.weight(1f)
+        modifier = if (chartHeight != null) {
+          modifier.height(chartHeight)
+        } else modifier.fillMaxHeight()
```

Comment:
This can be a chained modifier: 

```suggestion
        val baseModifier = Modifier.weight(1f)
        val modifier = if (chartHeight != null) {
          baseModifier.height(chartHeight)
        } else {
          baseModifier.fillMaxHeight()
        }
```

Reply:
Thanks, will fix 👍